tests/robustness/pathlock/path_lock.go

```diff
@@ -98,15 +137,14 @@ func (pl *PathLock) Unlock(path string) {
 
 // isInPath is a helper to determine whether one path is
 // either the same as another, or a child path (recursively) of it.
-func isInPath(path1, path2 string) bool {
+func isInPath(path1, path2 string) (bool, error) {
 	relFP, err := filepath.Rel(path2, path1)
 	if err != nil {
-		// Not sure - just wait anyway?
-		return true
+		return true, err
 	}
 
 	// If the relative path contains "..", this function will
 	// return false, because it is a cousin path. Only children (recursive)
 	// and the path itself will return true.
-	return !strings.Contains(relFP, "..")
+	return !strings.Contains(relFP, ".."), nil
```

Comment:
Can there be ".." present after the call to filepath.Abs()?

Reply:
Yes, `Rel` just finds the relative path required to change directory from one path to the other (if possible), e.g.:
https://play.golang.org/p/Q7W48tIYp70

Relative path from `/a/b/c` to `/a/b/d` is "up one directory to `/a/b` and down into `d`", i.e. `../d`